fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1380,57 +1384,45 @@ private ScoredPlan planText(@Nonnull CandidateScan candidateScan,
 
     @Nonnull
     private RecordQueryPlan planScan(@Nonnull CandidateScan candidateScan,
-                                     @Nullable ScanComparisons scanComparisons,
+                                     @Nonnull IndexScanParameters scanParameters,
                                      boolean strictlySorted) {
-        return planScan(candidateScan, null, scanComparisons, strictlySorted);
-    }
-
-    @Nonnull
-    private RecordQueryPlan planScan(@Nonnull CandidateScan candidateScan,
-                                     @Nullable IndexScanType scanType,
-                                     @Nullable ScanComparisons scanComparisons) {
-        return planScan(candidateScan, scanType, scanComparisons, false);
-    }
-
-    @Nonnull
-    private RecordQueryPlan planScan(@Nonnull CandidateScan candidateScan,
-                                     @Nullable IndexScanType scanType,
-                                     @Nullable ScanComparisons scanComparisons,
-                                     boolean strictlySorted) {
-        if (scanComparisons == null) {
-            scanComparisons = ScanComparisons.EMPTY;
-        }
         RecordQueryPlan plan;
         Set<String> possibleTypes;
         if (candidateScan.index == null) {
+            final ScanComparisons scanComparisons = ((IndexScanComparisons)scanParameters).getComparisons();
```

Comment:
Should the `planScan` method take an `IndesScanComparisons` in its signature to prevent this from being called with the wrong type of `IndexScanParameters`? Or will this change to handle more exotic scan parameter types as more are added?

Reply:
Since both times it is called with one I guess there is no reason not to do that for now.